### PR TITLE
Fix typing

### PR DESCRIPTION
**Pull request by kongzii**

### CODE REVIEW

File: agent.py [hunk 1]
```diff
@@ -32,7 +36,10 @@ def main(
             f"Found {len(available_markets)} markets: {[m.question for m in available_markets]}"
         )
 
-        market = agent.pick_market(available_markets)
+        market = t.cast(
+            t.Union[manifold.ManifoldMarket, omen.OmenMarket],
+            agent.pick_market(available_markets),
+        )  # TODO: Mypy bug: Works in VSCode type-checking, but doesn't in Mypy.
```
Comment (evangriffiths) [inline agent.py:42]: Does
```
market: t.Union[manifold.ManifoldMarket, omen.OmenMarket] = agent.pick_market(available_markets)
```
not work?

File: prediction_market_agent/tools/google_search.py [hunk 2]
```diff
@@ -4,7 +4,7 @@
 from prediction_market_agent import utils
 
 
-def google_search(query: str) -> t.List[str]:
```
Comment (evangriffiths) [inline prediction_market_agent/tools/google_search.py:7]: does mypy not like `t.List` or `t.Dict`?

File: prediction_market_agent/agents/abstract.py [hunk 3]
```diff
@@ -5,10 +5,7 @@
 
 
 class AbstractAgent:
-    def __init__(self):
-        pass
-
-    def pick_market(self, markets: list[MarketVar]) -> MarketVar:
+    def pick_market(self, markets: t.Sequence[MarketVar]) -> MarketVar:
```
Comment (evangriffiths) [inline prediction_market_agent/agents/abstract.py:8]: why this change?

File: prediction_market_agent/markets/omen.py [hunk 4]
```diff
@@ -440,11 +443,11 @@ def omen_sell_outcome_tx(
         web3=web3,
         market=market,
         approve=True,
-        from_address=from_address,
+        from_address=from_address_checksummed,
```
Comment (evangriffiths) [inline prediction_market_agent/markets/omen.py:446]: bugfix?

File: prediction_market_agent/tools/betting_strategies.py [hunk 5]
```diff
@@ -171,49 +184,53 @@ def _get_kelly_criterion_bet(
...
     max_bet: xDai,
 ) -> t.Tuple[xDai, OutcomeIndex]:
-    market: Market = get_market(market_address)
     if len(market.outcomeTokenAmounts) != 2:
         raise ValueError("Only binary markets are supported.")
 
-    # For FPMMs, the probability is equal to the marginal price
-    current_p_yes = Probability(market.outcomeTokenMarginalPrices[0])
-    outcome_index = 0 if estimated_p_yes > current_p_yes else 1
+    current_p_yes = market.outcomeTokenProbabilities[0]
```
Comment (evangriffiths) [inline prediction_market_agent/tools/betting_strategies.py:194]: ah nice

**Review by evangriffiths — APPROVED** [review]
Mammouth effort!

File: agent.py (hunk 1 above)
Comment (kongzii) [reply agent.py:42]: It doesn't :( I'm not sure what's Mypy's problem here, with Pylance (Pyright) it's just okay: 

Screenshot by Dropbox Capture

File: prediction_market_agent/tools/google_search.py (hunk 2 above)
Comment (kongzii) [reply prediction_market_agent/tools/google_search.py:7]: Sorry! It's my old habit to switch `t.X` to just `x[]`.

Do you prefer `t.List[..]`? Doing `list[...], set[...], dict[...]` is available from Python 3.9 without the need to import `typing`.

File: prediction_market_agent/agents/abstract.py (hunk 3 above)
Comment (kongzii) [reply prediction_market_agent/agents/abstract.py:8]: Just to make Mypy happy. The sequence is immutable, but I think using `list` should be OK in this case. But Mypy doesn't believe so 😄

File: prediction_market_agent/markets/omen.py (hunk 4 above)
Comment (kongzii) [reply prediction_market_agent/markets/omen.py:446]: Only type-fix, code worked OK. But `get_transaction_count` used somewhere along the way requires a checksummed-type address.

File: prediction_market_agent/tools/google_search.py (hunk 2 above)
Comment (evangriffiths) [reply prediction_market_agent/tools/google_search.py:7]: I guess I've been doing e.g. `t.List` instead of `list`, but I don't have a strong preference - we can just pick one. I'm happy to go with python types!

File: agent.py (hunk 1 above)
Comment (kongzii) [reply agent.py]: Not needed anymore, thanks to the removal of Protocol! 🎉
